### ISSUE TITLE
Windows: Building via devtools crashes - explicit build from git clone works

### ISSUE THREAD

**Issue by dmenne** [r-lib/testthat#1207]
Downloading the current github and building it locally works fine. `devtools::install_github` fails with compiler crash. Reproducably, multiple reboots between.

```
R version 4.0.2 (2020-06-22)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19041)
```

```
devtools::install_github("r-lib/testthat")
Using github PAT from envvar GITHUB_PAT
Downloading GitHub repo r-lib/testthat@HEAD
v  checking for file 'C:\Users\Dieter\AppData\Local\Temp\Rtmp6lsSpd\remotes329c51944fde\r-lib-testthat-12a39de/DESCRIPTION' (453ms)
-  preparing 'testthat': (5.3s)
v  checking DESCRIPTION meta-information ...
-  cleaning src
-  checking for LF line-endings in source and make files and shell scripts (766ms)
-  checking for empty or unneeded directories
-  building 'testthat_2.99.0.9000.tar.gz'

* installing *source* package 'testthat' ...
** using staged installation
** libs
"C:/rtools40/mingw64/bin/"gcc  -I"C:/R/include" -DNDEBUG -I../inst/include -DCOMPILING_TESTTHAT         -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign -c init.c -o init.o
"C:/rtools40/mingw64/bin/"gcc  -I"C:/R/include" -DNDEBUG -I../inst/include -DCOMPILING_TESTTHAT         -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign -c reassign.c -o reassign.o
"C:/rtools40/mingw64/bin/"g++ -std=gnu++11  -I"C:/R/include" -DNDEBUG -I../inst/include -DCOMPILING_TESTTHAT         -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign -c test-catch.cpp -o test-catch.o
"C:/rtools40/mingw64/bin/"g++ -std=gnu++11  -I"C:/R/include" -DNDEBUG -I../inst/include -DCOMPILING_TESTTHAT         -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign -c test-example.cpp -o test-example.o
Exception: STATUS_ACCESS_VIOLATION at rip=001800A5BFD
rax=0000000000030000 rbx=0000000000000002 rcx=0000000180320260
rdx=0000000000410000 rsi=0000000180320288 rdi=00000001005CF86C
r8 =00000000FFFFC89C r9 =0000000000000000 r10=0000000100000000
r11=0000000100429FB2 r12=00000000FFFFC930 r13=0000000180320260
r14=00000000FFFFCC80 r15=0000000180320EB8
rbp=000000000000002D rsp=00000000FFFFC810
program=C:\rtools40\usr\bin\sh.exe, pid 1991, thread main
cs=0033 ds=002B es=002B fs=0053 gs=002B ss=002B
make: *** [C:/R/etc/x64/Makeconf:222: init.o] Segmentation fault (core dumped)
make: *** Waiting for unfinished jobs....
Exception: STATUS_ACCESS_VIOLATION at rip=001800A5BFD
rax=0000000000030000 rbx=0000000000000002 rcx=0000000180320260
rdx=0000000000410000 rsi=0000000180320288 rdi=00000001005CF86C
r8 =00000000FFFFC89C r9 =0000000000000000 r10=0000000100000000
r11=0000000100429FB2 r12=00000000FFFFC930 r13=0000000180320260
r14=00000000FFFFCC80 r15=00000001803209F8
rbp=000000000000002D rsp=00000000FFFFC810
program=C:\rtools40\usr\bin\sh.exe, pid 1993, thread main
cs=0033 ds=002B es=002B fs=0053 gs=002B ss=002B
make: *** [C:/R/etc/x64/Makeconf:229: test-catch.o] Segmentation fault (core dumped)
make: *** [C:/R/etc/x64/Makeconf:222: reassign.o] Segmentation fault (core dumped)
make: *** [C:/R/etc/x64/Makeconf:229: test-example.o] Segmentation fault (core dumped)
ERROR: compilation failed for package 'testthat'
* removing 'C:/R/library/testthat'
Warnmeldungen:
1: In untar2(tarfile, files, list, exdir, restore_times) :
  skipping pax global extended headers
2: In untar2(tarfile, files, list, exdir, restore_times) :
  skipping pax global extended headers
3: In i.p(...) :
  installation of package 'C:/Users/Dieter/AppData/Local/Temp/Rtmp6lsSpd/file329c1c09e98/testthat_2.99.0.9000.tar.gz' had non-zero exit status
```

**Comment by gaborcsardi** [comment]
It seems that `sh.exe` crashes, which means that it is unlikely that this is a windows issue. My guess is that your rtools toolchain might be corrupt, or your machine has a memory module that is failing.

I would suggest
1. running some virus scanner against `sh.exe`. 
2. reinstalling R and rtools 4.x.
3. running some diagnostics for faulty RAM

**Comment by dmenne** [comment]
This is a developer machine used daily with devtools and Rcpp, but nevertheless I did everything you asked for. R now 4.03. Same result. This looks like out-of memory I sometimes seen when building under too small Docker memory, but this is running under plain Windows 24 GB.

I give up and build manually, which works.

**Comment by gaborcsardi** [comment]
You can try to compare the build flags that devtools/pkgbuild is passing to the compiler, maybe there is something there that triggers the bug.

**Comment by gaborcsardi** [comment]
FWIW I tried `remotes::install_github()` on `r-lib/testthat`, on Windows 10, in standalone mode (https://github.com/r-lib/remotes#standalone-mode), and it installed fine.

**Comment by dmenne** [comment]
I admit I did not know about this option because I rarely have problem when not using Docker. Too bad, with

`Sys.setenv(R_REMOTES_STANDALONE="true")`

same error. 

I do not know if Makevars and Makevars.win in ~./R are relevant, but I removed these. I have some nasty setting in that move like hell on every iteration of rstan, so I thought this could be a problem. Same result.

Never mind, must be some other settings. I do not care because can build via github clone easily.
